pkg/cmd/component/environment/add.go

```diff
@@ -0,0 +1,234 @@
+	component "github.com/redhat-developer/kam/pkg/pipelines/component"
+	hpipelines "github.com/redhat-developer/kam/pkg/pipelines/component"
+	env "github.com/redhat-developer/kam/pkg/pipelines/component/environment"
+	"github.com/redhat-developer/kam/pkg/pipelines/ioutils"
+	"github.com/spf13/cobra"
+
+	ktemplates "k8s.io/kubectl/pkg/util/templates"
+)
+
+const (
+	// AddEnvRecommendedCommandName the recommended command name
+	AddEnvRecommendedCommandName = "add"
+	componentNameFlag            = "component-name"
+	outputFolderNameFlag         = "output"
+	applicationNameFlag          = "application-name"
+	environmentNameFlag          = "env-name"
+
+	ApplicationNameAddEnvironmentMessage = "Provide the Application name to add an Environment"
+)
+
+var (
+	addEnvExample = ktemplates.Examples(`
+	# Add a new environment to the application's component in the GitOps repository
+	# Example: kam env add --env-name new-env --output <path to GitOps folder>
+	
+	%[1]s 
+	`)
+
+	addEnvLongDesc  = ktemplates.LongDesc(`Add a new environment to the application's component in the GitOps repository`)
+	addEnvShortDesc = `Add a new environment`
+)
+
+// AddEnvParameters encapsulates the parameters for the kam pipelines init command.
+type AddEnvParameters struct {
+	*hpipelines.GeneratorOptions
+	Interactive bool
+}
+
+// NewAddEnvParameters bootstraps a AddEnvParameters instance.
+func NewAddEnvParameters() *AddEnvParameters {
+	return &AddEnvParameters{
+		GeneratorOptions: &hpipelines.GeneratorOptions{},
+	}
+}
+
+// Checking the mandatory flags & reusing missingFlagErr in .go
+func checkMandatoryFlags(flags map[string]string) error {
+	missingFlags := []string{}
+	mandatoryFlags := []string{componentNameFlag, applicationNameFlag, environmentNameFlag}
+	for _, flag := range mandatoryFlags {
+		if flags[flag] == "" {
+			missingFlags = append(missingFlags, fmt.Sprintf("%q", flag))
+		}
+	}
+	if len(missingFlags) > 0 {
+		return missingFlagErr(missingFlags)
+	}
+	return nil
+}
+
+func missingFlagErr(flags []string) error {
+	return fmt.Errorf("required flag(s) %s not set", strings.Join(flags, ", "))
+}
+
+func initiateNonInteractiveModeComponent(io *AddEnvParameters) error {
+	appFs := ioutils.NewFilesystem()
+	mandatoryFlags := map[string]string{componentNameFlag: io.ComponentName, applicationNameFlag: io.ApplicationName, environmentNameFlag: io.EnvironmentName}
+	if err := checkMandatoryFlags(mandatoryFlags); err != nil {
+		return err
+	}
+	err := ui.ValidateName(io.ApplicationName)
+	if err != nil {
+		return err
+	}
+	err = ui.ValidateName(io.ComponentName)
+	if err != nil {
+		return err
+	}
+	if io.Output == "./" {
+		path, _ := os.Getwd()
+		io.Output = path
+	}
+	exists, _ := ioutils.IsExisting(appFs, io.Output)
+	if !exists {
+		return fmt.Errorf("the provided path  %s does not exist. ", io.Output)
+	}
+	return nil
+}
+
+func initiateInteractiveModeComponent(io *AddEnvParameters, cmd *cobra.Command) error {
```

Comment:
Hey !
When we run the `kam env add` it starts a interactive prompt without asking for default values. 
If user wants to add `--output` flag we cannot do it.